menu/menu.lua

```diff
@@ -15,21 +15,71 @@ menu.up = function()
     if menu.mapSelect > #mapList then
         menu.mapSelect = 1
     end
+
+    menu.threeBest = timer.getBestTimes(menu.mapSelect, 3)
 end
 
 menu.down = function()
     menu.mapSelect = menu.mapSelect - 1
     if menu.mapSelect < 1 then
         menu.mapSelect = #mapList
     end
+
+    menu.threeBest = timer.getBestTimes(menu.mapSelect, 3)
+end
+
+local function displayTime()
+    -- Draw current time and best times.
+    local isNewBest = false
+    local convertedTime
+    local corner = { State.camera.pos_x, State.camera.pos_y }
+
+    -- Display either the last finish time (if map has finished) or current time.
+    if State.activeMap ~= -1 then
+        convertedTime = timer.convertSeconds(State.seconds)
+    else
+        convertedTime = timer.convertSeconds(State.lastCompletedTime)
+    end
+
+    -- Display three best times on record
+    for i = 0, 2 do
+        -- If map was completed and time is one of the best, display it in gold.
+        if menu.threeBest[i + 1] == convertedTime then
+            isNewBest = true
+            love.graphics.setColor(255, 215, 0, 1)
+        end
+        -- Print out times
+        love.graphics.print('#' .. i + 1 .. ': ' .. menu.threeBest[i + 1],
+                            corner[1] + 448, corner[2] + 430 + (i * 45), 0, 1.5, 1.5)
```

Comment:
Love GUI formatting math. Very much more elegant than mine

Reply:
Yes I was particularly happy with that one. I live for little tricks like that.